### PR TITLE
Tweak autoconnect wallet behavior

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts [hunk 1]
```diff
@@ -1,51 +1,62 @@
...
 
-import { useEffect } from 'react';
+import { useQuery } from '@tanstack/react-query';
 
 import { useConnectWallet } from './useConnectWallet.js';
 import { useCurrentWallet } from './useCurrentWallet.js';
 import { useWallets } from './useWallets.js';
 import { useWalletStore } from './useWalletStore.js';
 
 export function useAutoConnectWallet(autoConnectEnabled: boolean) {
```
Comment (hayes-mysten) [inline sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts:11]: should this default to true?

**Review by hayes-mysten — APPROVED** [review]

File: sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts [hunk 2]
```diff
@@ -1,51 +1,62 @@
...
-	useEffect(() => {
-		if (
-			!autoConnectEnabled ||
-			!lastConnectedWalletName ||
-			!lastConnectedAccountAddress ||
-			!isDisconnected
-		) {
-			return;
-		}
+	useQuery({
```
Comment (williamrobertson13) [inline sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts:18]: I'm not sure I see the benefit here between having this be a query with the sort of `useIsFetching` / `queryCache` code versus keeping it as a mutation with `useMutationState`. This query isn't cached and has no retries, so doesn't this effectively the same as what we previously had with the mutation in the `useEffect`?

I think I might just be confused about the vision here haha, but it feels "off" to have this be a query that is performing a side effect and returns data that can't be accessed by normal conventions

Comment (Jordan-Mysten) [reply sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts:18]: The query is cached?

File: sdk/dapp-kit/src/hooks/wallet/useAutoConnectionStatus.ts [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
 import { useWalletStore } from './useWalletStore.js';
 
 /**
  * Retrieves the status for the initial wallet auto-connection process.
  */
-export function useAutoConnectionStatus() {
-	// TODO: Replace this with shareable mutation state once we require react-query v5
-	return useWalletStore((state) => state.autoConnectionStatus);
+export function useAutoConnectionStatus(): 'disabled' | 'idle' | 'attempted' {
```
Comment (williamrobertson13) [inline sdk/dapp-kit/src/hooks/wallet/useAutoConnectionStatus.ts:12]: can we update the docs for the `settled` -> `attempted` change and also maybe change the description for those states to be clearer (IIRC, I don't think what I originally wrote accurately reflects this behavior)

**Review by williamrobertson13 — APPROVED** [review]
good stuff!

File: sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts (hunk 1 above)
Comment (Jordan-Mysten) [reply sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts]: RE defaulting to true, I think we've decided at least for now to disable autoconnect to false, but would like to reconsider.

File: sdk/dapp-kit/src/hooks/wallet/useAutoConnectionStatus.ts (hunk 3 above)
Comment (Jordan-Mysten) [reply sdk/dapp-kit/src/hooks/wallet/useAutoConnectionStatus.ts:12]: Yeah updated the docs!

**Pull request merged**